packages/components/src/utils/events.ts

```diff
@@ -0,0 +1,51 @@
+type EventHandler< T extends Event > = ( event: T ) => void;
+
+/**
+ * Merges event handlers together.
+ *
+ * @template TEvent
+ * @param  handler
+ * @param  otherHandler
+ */
+function mergeEvent< TEvent extends Event >(
+	handler: EventHandler< TEvent >,
+	otherHandler: EventHandler< TEvent >
+): EventHandler< TEvent > {
+	return ( event: TEvent ) => {
+		if ( typeof handler === 'function' ) {
+			handler( event );
+		}
+		if ( typeof otherHandler === 'function' ) {
+			otherHandler( event );
+		}
+	};
+}
+
+/**
+ * Merges a set of event handlers together.
+ *
+ * @template TEvent
+ * @param  handlers
+ * @param  extraHandlers
+ */
+export function mergeEventHandlers<
+	TEvent extends Event,
+	TLeft extends Record< string, EventHandler< TEvent > >
+>(
+	handlers: TLeft,
+	extraHandlers: Record< string, EventHandler< TEvent > >
+): TLeft {
+	const mergedHandlers: TLeft = { ...handlers };
+
+	for ( const [ key, handler ] of Object.entries( mergedHandlers ) ) {
+		if ( typeof extraHandlers[ key ] === 'function' ) {
```

Comment:
What about the keys that are only present in `extraHandlers` ? As of now, they seem to be ignored, instead of being merged.

I think that we should loop over `extraHandlers` too, and only then check if the current event key already exist

Reply:
Yes as I said in https://github.com/WordPress/gutenberg/pull/33205#issuecomment-875581202 I think we should do that?

Reply:
Unless there are usages in the `TextInput` component that clearly indicate otherwise, I would go ahead and make changes so that all properties of `extraHandlers` are present in the returned value

Reply:
This was addressed in https://github.com/WordPress/gutenberg/pull/33205/commits/a41f480751de0bb41a9f186266c90435817f706d